#! /bin/sh /usr/share/dpatch/dpatch-run
## 20_watch_8bitchar.dpatch by  <csmall@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Make char unsigned

@DPATCH@
diff -urNad procps-3.2.8~/watch.c procps-3.2.8/watch.c
--- procps-3.2.8~/watch.c	2009-05-11 22:20:14.000000000 +1000
+++ procps-3.2.8/watch.c	2009-05-11 22:22:52.000000000 +1000
@@ -297,7 +297,7 @@
 				move(y, x);
 				if (option_differences) {
 					chtype oldch = inch();
-					char oldc = oldch & A_CHARTEXT;
+					unsigned char oldc = oldch & A_CHARTEXT;
 					attr = !first_screen
 					    && ((char)c != oldc
 						||
